youtube_dl/extractor/theplatform.py

```diff
@@ -37,9 +37,14 @@ def _extract_theplatform_smil(self, smil_url, video_id, note='Downloading SMIL d
             smil_url, video_id, note=note, query={'format': 'SMIL'},
             headers=self.geo_verification_headers())
         error_element = find_xpath_attr(meta, _x('.//smil:ref'), 'src')
-        if error_element is not None and error_element.attrib['src'].startswith(
-                'http://link.theplatform.com/s/errorFiles/Unavailable.'):
-            raise ExtractorError(error_element.attrib['abstract'], expected=True)
+        try:
+            if error_element is not None and error_element.attrib['src'].startswith(
+                    'http://link.theplatform.com/s/errorFiles/Unavailable.'):
+                raise ExtractorError(error_element.attrib['abstract'], expected=True)
+        except ExtractorError:
+            if error_element.attrib['title'].startswith('No AssetType'):
+                return [], {}
+            raise
```

Comment:
This does not make any sense. You must **catch** exceptions **where** you use `_extract_theplatform_smil` not here.